### PR TITLE
Extend orderbook test to include swap execution

### PR DESCRIPTION
**Pull request by rishflab**
Extended the orderbook test to execute the swap created from an order.

Various bug fixes in comit, cnd and the test code to get the extended test working.

There will be a follow up PR to cleanup the test code to make it more readable.

### CODE REVIEW

File: comit/src/network/protocols/orderbook.rs [hunk 1]
```diff
@@ -450,7 +450,7 @@ mod tests {
             (
                 BehaviourOutEvent::TakeOrderConfirmation {
                     peer_id: alice_got_peer_id,
-                    order_id: alice_got_order_id,
+                    order_id: _,
```
Comment (tcharding) [inline comit/src/network/protocols/orderbook.rs:453]: I prefer
```
commit 3aaceb8e98d1d2d0c3c1a5a3f3212cb99e21c582
Author: Tobin C. Harding <tobin@coblox.tech>
Date:   Fri Jul 17 13:57:08 2020 +1000

    Add exhaustive assertions to orderbook unit test

diff --git a/comit/src/network/protocols/orderbook.rs b/comit/src/network/protocols/orderbook.rs
index 6d0caf43f..f5e8e90c3 100644
--- a/comit/src/network/protocols/orderbook.rs
+++ b/comit/src/network/protocols/orderbook.rs
@@ -461,7 +461,10 @@ mod tests {
             ) => {
                 assert_eq!(alice_got_peer_id, bob.peer_id);
                 assert_eq!(bob_got_peer_id, alice.peer_id);
-                assert_eq!(bob_got_order_id, alice_order.id);
+
+                assert_eq!(alice_got_order_id, alice_order.id);
+                assert_eq!(bob_got_order_id, alice_got_order_id);
+
                 assert_eq!(alice_got_swap_id, bob_got_swap_id);
             }
             _ => panic!("failed to get take order confirmation"),
```
I can push this if you like

Comment (tcharding) [reply comit/src/network/protocols/orderbook.rs:453]: I'll push it later on, after this merges.

Comment (rishflab) [reply comit/src/network/protocols/orderbook.rs:453]: bors r+

**Comment by tcharding** [comment]
Oh it looks like bors parsed my initial message suggesting to message it.

**Pull request merged**